server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/service/ExportService.java

```diff
@@ -0,0 +1,93 @@
+import java.io.PipedInputStream;
+import java.io.PipedOutputStream;
+
+import org.apache.james.blob.api.BlobId;
+import org.apache.james.blob.api.BlobStore;
+import org.apache.james.blob.export.api.BlobExportMechanism;
+import org.apache.james.blob.export.api.FileExtension;
+import org.apache.james.core.Username;
+import org.apache.james.mailbox.backup.MailboxBackup;
+import org.apache.james.task.Task;
+import org.apache.james.user.api.UsersRepository;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.fge.lambdas.Throwing;
+
+import reactor.core.publisher.Mono;
+
+// TODO Move me to WebAdminMailboxesBackUp
+public class ExportService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ExportService.class);
+    private static final String EXPLANATION = "The backup of your mailboxes has been shared with you";
+    private static final String FILE_PREFIX = "mailbox-backup-";
+
+    private final MailboxBackup mailboxBackup;
+    private final BlobStore blobStore;
+    private final BlobExportMechanism blobExport;
+    private final UsersRepository usersRepository;
+
+    public ExportService(MailboxBackup mailboxBackup, BlobStore blobStore, BlobExportMechanism blobExport, UsersRepository usersRepository) {
+        this.mailboxBackup = mailboxBackup;
+        this.blobStore = blobStore;
+        this.blobExport = blobExport;
+        this.usersRepository = usersRepository;
+    }
+
+    public Mono<Task.Result> exportUserMailboxesData(Username username) {
+        return Mono.using(
+            () -> zipMailboxesContent(username),
+            inputStream -> blobStore.save(blobStore.getDefaultBucketName(), inputStream, BlobStore.StoragePolicy.LOW_COST)
+                    .flatMap(blobId -> export(username, blobId)
+                        .then(blobStore.delete(blobStore.getDefaultBucketName(), blobId))
+                        .thenReturn(Task.Result.COMPLETED))
+                    .onErrorResume(e -> {
+                        LOGGER.error("Error exporting mailboxes of user: {}", username.asString(), e);
+                        return Mono.just(Task.Result.PARTIAL);
+                    }),
+            Throwing.consumer(InputStream::close).sneakyThrow(), false);
+    }
+
+    private Mono<Void> export(Username username, BlobId blobId) {
+        return Mono.fromRunnable(Throwing.runnable(() ->
+            blobExport.blobId(blobId)
+                .with(usersRepository.getMailAddressFor(username))
+                .explanation(EXPLANATION)
+                .filePrefix(FILE_PREFIX + username.asString() + "-")
+                .fileExtension(FileExtension.ZIP)
+                .export()));
+    }
+
+    private InputStream zipMailboxesContent(Username username) throws Exception {
+        PipedOutputStream out = new PipedOutputStream();
+        PipedInputStream in = new PipedInputStream();
+        out.connect(in);
+        mailboxBackup.backupAccount(username, out);
```

Comment:
IMO we need to schedule this in another thread upon resource creation...

```
Mono.using(
        () -> out,
        out -> Mono.fromRunnable(() -> mailboxBackup.backupAccount(username, out)),
        out::close)
    .subscribeOn(Schedulers.elastic())
    .subscribe();
```

Otherwize we synchronously run the pipe which will likely end up in a buffer, and that's not the intent...